File: convertedToIOS5renamed/ch33p848map/p707p723map/MyAnnotation.m


#import "MyAnnotation.h"


@implementation MyAnnotation

@synthesize coordinate, title, subtitle;


- (id)initWithLocation: (CLLocationCoordinate2D) coord {
    self = [super init];
    if (self) {
        self->coordinate = coord;
    }
    return self;
}

@end

File: bk2ch21p744maps/ch34p1008maps/MyAnnotation.swift


import UIKit
import MapKit

class MyAnnotation : NSObject, MKAnnotation {

//@property (nonatomic, readwrite) CLLocationCoordinate2D coordinate;
    
    let coordinate : CLLocationCoordinate2D
    var title : String!
    var subtitle : String!
    
    init(location coord:CLLocationCoordinate2D) {
        self.coordinate = coord
        super.init()
    }
    
//@property (nonatomic, copy) NSString *title, *subtitle;
//- (id)initWithLocation:(CLLocationCoordinate2D)coord;

}